dot/core/state.go

```diff
@@ -34,13 +34,26 @@ type BlockState interface {
 	SubChain(start, end common.Hash) ([]common.Hash, error)
 	AddBlock(*types.Block) error
 	AddBlockWithArrivalTime(*types.Block, uint64) error
-	SetBlock(*types.Block) error
+
+	SetBlockBody(common.Hash, *types.Body) error
+	GetBlockBody(common.Hash) (*types.Body, error)
+
+	CompareAndSetBlockData(bd *types.BlockData) error
+
 	SetHeader(*types.Header) error
 	GetHeader(common.Hash) (*types.Header, error)
+
+	GetReceipt(common.Hash) ([]byte, error)
+	SetReceipt(common.Hash, []byte) error
```

Comment:
I don't see the `SetReceipt`, `SetMessageQueue` or `SetJustification` functions used anywhere in core, please remove if unused

Reply:
done